### PR TITLE
Fixed issue- #4645 | Use YAML configuration for java  AbstractEncrypt…

### LINKED ISSUE #4645: Use YAML configuration instead of java configuration for DataSource test
YAML configuration is clear and easy to understand. 
Using YAML configuration to instead of java configuration for DataSource test is better.
The test cases need to be refactored:

- [ ] AbstractEncryptJDBCDatabaseAndTableTest for encrypt rule creation
- [ ] AbstractMasterSlaveJDBCDatabaseAndTableTest for master slave rule creation
- [ ] AbstractShardingJDBCDatabaseAndTableTest for sahrding rule creation

### PR DESCRIPTION
**Pull request by ranjeet-floyd**
Fixes #4645.

Changes proposed in this pull request:
-  Used YAML configuration instead of java configuration for AbstractEncryptJDBCDatabaseAndTableTest.java

### CODE REVIEW

File: sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java [hunk 1]
```diff
@@ -30,93 +41,103 @@
...
-import java.util.Collections;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Properties;
+import org.yaml.snakeyaml.Yaml;
 
 public abstract class AbstractEncryptJDBCDatabaseAndTableTest extends AbstractSQLTest {
-    
+
```
Comment (kimmking) [inline sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java]: should keep 4 blankspace

File: sharding-jdbc/sharding-jdbc-core/src/test/resources/encrypt_data.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    plainColumn: "plain_pwd"
+    cipherColumn: "cipher_pwd"
+    assistedQueryColumn: ""
+    encryptor: "test"
+t_encrypt2:
+    plainColumn: "plain_pwd2"
+    cipherColumn: "cipher_pwd2"
+    assistedQueryColumn: ""
+    encryptor: "test"
+
```
Comment (kimmking) [inline sharding-jdbc/sharding-jdbc-core/src/test/resources/encrypt_data.yaml]: only one empty line at the tail.

**Comment by ranjeet-floyd** [comment]
Got it, will update PR .

File: sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java (hunk 1 above)
Comment (ranjeet-floyd) [reply sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java]: sorry , can't get you . I used project check-style config.

Comment (kimmking) [reply sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java]: plz let origin style here with 4 blankspaces

Comment (ranjeet-floyd) [reply sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java]: Used same checkstyle. Do you any difference.

File: sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java [hunk 3]
```diff
@@ -17,87 +17,90 @@
...
         encryptDataSourceWithProps = new EncryptDataSource(dataSources.values().iterator().next(), new EncryptRule(createEncryptRuleConfiguration()), createProperties());
     }
-    
+
     private static Properties createProperties() {
         Properties result = new Properties();
         result.put(ConfigurationPropertyKey.SQL_SHOW.getKey(), true);
         result.put(ConfigurationPropertyKey.QUERY_WITH_CIPHER_COLUMN.getKey(), false);
         return result;
     }
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java]: Please use YAML to config properties too.

File: sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java [hunk 4]
```diff
@@ -17,87 +17,90 @@
...
+                EncryptColumnRuleConfiguration columnRuleConfiguration = new EncryptColumnRuleConfiguration(yamlConfig.getPlainColumn(), yamlConfig.getCipherColumn(),
+                    yamlConfig.getAssistedQueryColumn(),
+                    yamlConfig.getEncryptor());
+                columns.put(columnMap.getKey(), columnRuleConfiguration);
+            }
+            result.getTables().put(tableMap.getKey(), new EncryptTableRuleConfiguration(columns));
+        }
+
         return result;
     }
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java]: You may use `YamlEncryptDataSourceFactory.createDataSource()` to load YAML directly.
We can just make sure the format of YAML file are correct and then use ShardingSphere's public API to load yaml and create EncryptDataSource.

File: sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java (hunk 3 above)
Comment (ranjeet-floyd) [reply sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java]: Added properties in yaml config

File: sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java (hunk 4 above)
Comment (ranjeet-floyd) [reply sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java]: @terrymanu  Modified as per comment. Pls check.

**Comment by terrymanu** [comment]
@ranjeet-floyd  Thank you, Please fix CI.

**Pull request merged**